File: server/dotnet/sdk-dotnet-example/Presentation/Validators/PaymentLinkValidators/PaymentLinkValidator.cs
using Business.Domain.Common.Enums;
using Business.Domain.Enums.PaymentLinks;
using FluentValidation;
using Presentation.Models.CreatePaymentLinks;
using Action = Business.Domain.Enums.PaymentLinks.Action;

namespace Presentation.Validators.PaymentLinkValidators;

public class PaymentLinkValidator : AbstractValidator<CreatePaymentLinkRequest>
{
    public PaymentLinkValidator()
    {
        ValidatorOptions.Global.DefaultClassLevelCascadeMode = CascadeMode.Stop;

        RuleFor(x => x.Amount)
            .NotEmpty()
            .WithMessage("The Amount field is required.")
            .GreaterThan(0)
            .WithMessage("The Amount field must be greater than zero.");
        
        RuleFor(x => x.Description)
            .MaximumLength(1000)
            .When(x => !string.IsNullOrEmpty(x.Description))
            .WithMessage("Description field must be shorter than 1000 characters.");

        RuleFor(x => x.Currency)
            .NotEmpty()
            .WithMessage("The Currency field is required.")
            .IsInEnum()
            .WithMessage("The Currency field is invalid.")
            .Must(c => c is Currency.EUR or Currency.USD)
            .WithMessage("The Currency field is invalid.");

        RuleFor(x => x.ValidFor)
            .NotEmpty()
            .WithMessage("The ValidFor field is required.")
            .Must(c => c is ValidityPeriod.OneDay 
                or ValidityPeriod.TwoDays 
                or ValidityPeriod.TwoWeeks 
                or ValidityPeriod.OneMonth)
            .WithMessage("The ValidFor field is invalid and must be a number from the following set: 24, 48, 336, 720.");

        RuleFor(x => x.Action)
            .NotEmpty()
            .WithMessage("The Action field is required.")
            .IsInEnum()
            .WithMessage("The Action field is invalid.")
            .Must(c => c is Action.CREATE or Action.PREVIEW).WithMessage("The Action field is invalid.");
    }
}

File: server/nodejs/src/presentation/validators/payment-link/payment-link-validator.ts
import { ValidationError } from '../../../business/errors/validation-error';
import { Action } from '../../../business/domain/enums/payment-links/action';
import { ValidFor } from '../../../business/domain/enums/payment-links/validity-period';
import { Currency } from '../../../business/domain/common/enums/currency';
import { CreatePaymentLinkRequest } from '../../models/payment-link/create-payment-link-request';
import { EnumUtils } from '../../utils/enum-utils';

export class PaymentLinkValidator {
    static validate(request: CreatePaymentLinkRequest): string[] {
        const errors: string[] = [];

        if (request.amount === undefined || request.amount === null) {
            errors.push('The Amount field is required.');
        } else if (!Number.isFinite(Number(request.amount))) {
            errors.push('The Amount field must be a number.');
        } else if (Number(request.amount) <= 0) {
            errors.push('The Amount field must be greater than zero.');
        }

        if (!request.currency) {
            errors.push('The Currency field is required.');
        } else if (!EnumUtils.toEnum(Currency, request.currency)) {
            errors.push('The Currency field is invalid.');
        }

        if (request.description !== undefined && request.description !== null) {
            if (request.description.length > 1000) {
                errors.push('The Description field must be shorter than 1000 characters.');
            }
        }

        if (request.validFor === null || request.validFor === undefined) {
            errors.push('The ValidFor field is required.');
        } else {
            const validForStr = String(request.validFor);

            if (isNaN(Number(validForStr))) {
                errors.push('The ValidFor field is invalid.');
            } else if (!EnumUtils.toEnum(ValidFor, validForStr)) {
                errors.push(
                    'The ValidFor field is invalid and must be a number from the following set: 24, 48, 336, 720.',
                );
            }
        }

        if (!request.action) {
            errors.push('The Action field is required.');
        } else if (!EnumUtils.toEnum(Action, request.action)) {
            errors.push('The Action field is invalid.');
        }

        if (errors.length > 0) {
            throw new ValidationError(errors);
        }

        return errors;
    }
}
